src/librbd/journal/Types.h

```diff
@@ -121,12 +121,13 @@ struct OpFinishEvent : public OpEventBase {
 
 struct SnapEventBase : public OpEventBase {
   std::string snap_name;
+  cls::rbd::SnapshotNamespace snap_namespace;
```

Comment:
Yes, but the creation of new images should be a clone operation (that's RBD's core design win for OpenStack). That implies that there will be lots of edge cases to handle OpenStack since OpenStack won't treat the images as dependent. We can tackle that issue later.